Week 6/7 SW Drive + Mech (CV Integration and Auto Arch)

beranki opened the pull request:
Localization
CV Alignment with AprilTag (finds and drives to pose, works with and without constant vision.)
Added Slew Rate to TeleOp Drive
Patched Perspective Issues in Auto and TeleOp
LED FSM is added
Mech Logic Patches from Testing

beranki committed:
wip

beranki committed:
[wip] integrating commands into paths and separating reef and station
alignment in teleop.

beranki committed:
resolving merge

beranki committed:
resolving merg w/ main

beranki committed:
fixing checkstyles

beranki committed:
readding rpi

beranki committed:
adding absolute position apriltag json file

vismaybhargav committed:
(feat) wip move to new auto scheduling arch

beranki committed:
wip

beranki committed:
sim system revamped

beranki committed:
slight auto system revamped

beranki committed:
[add] revamped auto system, works with partial systems and with sim
sync.

beranki committed:
added tag align

beranki committed:
fixing checkstyles

beranki committed:
removing unnecessary function calls

Adbritto committed:
fix bug with funnrl commands

beranki committed:
logging now works

Adbritto committed:
merge with main and update latest mech changes

beranki committed:
added new path alterations

asuresh952 committed:
Editing Paths + Making new ones

asuresh952 committed:
Merge branch 'dev/week-5/sw-drive/auto-patch' of https://github.com/Tino-FRC-2473/FRC2025 into dev/week-5/sw-drive/auto-patch
Fixing auto paths, other code unchanged

asuresh952 committed:
Forgot one auto path whoops

beranki committed:
resolving brief mc

beranki committed:
completing rem merge

beranki committed:
modifed traj's, alliance flipping, passive fwd for alignment.

beranki committed:
fixing checkstyles

beranki committed:
wip - sign flipping for reef patched

beranki committed:
wip - add slew rate limiter

nigelpv committed:
(add) Half paths for pre allignment at ree

beranki committed:
testing + tuning changes

beranki committed:
Merge branch 'dev/week-6/sw-drive/cv-test' of https://github.com/Tino-FRC-2473/FRC2025 into dev/week-6/sw-drive/cv-test

beranki committed:
scaling clamp up

beranki committed:
cleaning up clamp / scaling

beranki committed:
slight tuning

beranki committed:
tweaked deadbands to allow more responsive driving + fixed slew rate and
rotation bug

beranki committed:
Merge branch 'dev/week-6/sw-drive/cv-test' of https://github.com/Tino-FRC-2473/FRC2025 into dev/week-6/sw-drive/cv-test

beranki committed:
adding some changes

beranki committed:
Field Testing Changes

Tuned CV Poses
Tested Elevator and Climb
Tested Manual Driving
Tested Teleop Cycles w/o cv

beranki committed:
wip localization -- added constants and profiling

beranki committed:
localization works, drive to pose wip

beranki committed:
basic p loop added

beranki committed:
testing changes

beranki committed:
localization works, position is confident but rotation is iffy; more tuning

beranki committed:
wip - testing

beranki committed:
added clamp back

beranki committed:
removed clamp w/ old constants - 0.01 deadband

beranki committed:
Merge https://github.com/Tino-FRC-2473/FRC2025 into dev/week-7/sw-drive/cv-localization

KGupta10 committed:
sort pose_list by hypotenuse before sending over NT

beranki committed:
fixing merge Mess

beranki committed:
Merge branch 'dev/week-7/sw-drive/cv-local' of https://github.com/Tino-FRC-2473/FRC2025 into dev/week-7/sw-drive/cv-local

fixing merge issues

beranki committed:
wip

beranki committed:
localization w/ camera rotation, and built in filter

beranki committed:
for auto, rotation sometimes goes the wrong way?

beranki committed:
more tuning and fixes

vismaybhargav committed:
testing changes

beranki committed:
wip

beranki committed:
wip

beranki committed:
THESE CONSTANTS WORK VERY WELL ON SIM

beranki committed:
all testing / sim perspective changes ironed out.

beranki committed:
slight changes + tuning paths

Adbritto committed:
use alliance orientation in drive to pose to prevent driving reverse on red alliance

Adbritto committed:
merge latest mech and led code

beranki committed:
got rid of alliance multiplier, autos and teleop are all now right direction.

Adbritto committed:
Merge branch 'dev/week-7/sw-drive/cv-local' of https://github.com/Tino-FRC-2473/FRC2025 into dev/week-7/sw-drive/cv-local

Adbritto committed:
update wpilib

beranki committed:
wip

beranki committed:
working alignment(prolly)

beranki committed:
WORKING CV ALIGNMETN AND LOCALIZATION !!!!

beranki committed:
fixing checkstyles

beranki committed:
fr fixing checkstyles

Adbritto committed:
Change LED Patterns

Adbritto committed:
merge mech changes

Adbritto committed:
Merge branch 'dev/week-7/sw-drive/cv-local' of https://github.com/Tino-FRC-2473/FRC2025 into dev/week-7/sw-drive/cv-local

beranki committed:
wip

beranki committed:
added current check function + int reset in teleop. to be merged with changes on laptop 3

Adbritto committed:
Merge branch 'main' into dev/week-7/sw-drive/cv-local

beranki committed:
tested constants for reef alignment

beranki committed:
added current limit

beranki committed:
station alignment changes

beranki committed:
field changes

beranki committed:
wip testing changes

beranki committed:
field testinng fixes

beranki committed:
Merge branch 'dev/week-7/sw-drive/cv-local' of https://github.com/Tino-FRC-2473/FRC2025 into dev/week-7/sw-drive/cv-local

merging test changes w/ driver requests

beranki committed:
new teleop elastic layout

Adbritto committed:
Merge branch 'dev/week-7/sw-drive/cv-local' of https://github.com/Tino-FRC-2473/FRC2025 into dev/week-7/sw-drive/cv-local

Adbritto committed:
fix checkstyles and add null checks for drivefsm

beranki committed:
added optimal reef alignment range marker - wip

beranki committed:
range wip

asuresh952 committed:
Finished Half Paths + Standardized them all, every half path completed.

asuresh952 committed:
Merge branch 'dev/week-7/sw-drive/cv-local' of https://github.com/Tino-FRC-2473/FRC2025 into dev/week-7/sw-drive/cv-local

asuresh952 committed:
Minor path fix

asuresh952 committed:
Fixed path name

queerboi committed:
update Funnel FSM diagram

queerboi committed:
Merge branch 'dev/week-7/sw-drive/cv-local' of https://github.com/Tino-FRC-2473/FRC2025 into dev/week-7/sw-drive/cv-local

Coldslicer committed:
fix issue with climber not going to lowered

Coldslicer committed:
fix error in previous commit

queerboi committed:
update climber FSM diagram

queerboi committed:
remove unused mech code

queerboi commented on src/main/java/frc/robot/HardwareMap.java:
```diff
@@ -18,11 +18,11 @@ public final class HardwareMap {
 	public static final int FUNNEL_BREAK_BEAM_DIO_PORT = 2; // FINAL
 
-	public static final int CLIMBER_LIMIT_SWITCH_DIO_PORT = 5; // for testing only
+	public static final int CLIMBER_LIMIT_SWITCH_DIO_PORT = 3; // for testing only
```
Is this finalized?

Adbritto replied:
yes

beranki committed:
testing changes

beranki committed:
Merge branch 'dev/week-7/sw-drive/cv-local' of https://github.com/Tino-FRC-2473/FRC2025 into dev/week-7/sw-drive/cv-local

merging climber changes and testing changes

queerboi commented on src/main/java/frc/robot/TeleopInput.java:
```diff
@@ -89,13 +89,37 @@ public boolean getDriveCircleButton() {
 	 * @return Axis value
 	 */
 	public boolean getDriveBackButtonPressed() {
```
I'd recommend renaming these getters to what they're used for, so like "getDriveSeedGyroButton()"

queerboi committed:
fix teleopinput javadocs

queerboi committed:
clean up mech code

queerboi commented on src/main/java/frc/robot/systems/ElevatorFSMSystem.java:
```diff
@@ -419,21 +435,22 @@ protected Distance getTarget() {
 		@Override
 		public boolean isFinished() {
-			return timer.get() > targTime;
+			return timer.get() > 1;
```
can we make this 1 an instance variable so that WaitCommand can wait for custom times?

Adbritto replied:
I was planning on removing the wait command, it was only temporary for testing

queerboi committed:
fix typos

beranki committed:
WORKING REEF ALIGN AND SCORE IN AUTO!!!! :fire: :goat:

AG6GR commented on src/main/java/frc/robot/CommandSwerveDrivetrain.java:
```diff
@@ -147,4 +151,83 @@ private void setupSimulation(Pose2d startingPose) {
+	}
+
+	// /**
+	//  * Configure the auto builder.
```
Please clean up unused code.

AG6GR commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -133,10 +148,10 @@ public void autonomousInit() {
-				driveSystem.getMapleSimDrivetrain().getDriveSimulation()
-					.setSimulationWorldPose(autoRoutines.getInitialAutoPose());
-			}
+			// if (Robot.isSimulation()) {
```
Do we need this?

Adbritto replied:
For sim purposes, yea

AG6GR commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -263,13 +293,28 @@ public void simulationPeriodic() {
+		if (driveSystem != null) {
+			driveSystem.updateLogging();
+			// if (HardwareMap.isCVHardwarePresent()) {
+				//driveSystem.updateVisionEstimates();
```
Do we need this?

AG6GR commented on src/main/java/frc/robot/auto/AutoRoutines.java:
```diff
@@ -47,16 +46,6 @@ private Command checkDriveCommands(AutoCommands commandEntry) {
 		switch (commandEntry) {
 			case DRIVE_BRAKE_CMD:
 				return driveSystem.brakeCommand();
```
Why is there only one command?

Adbritto replied:
All other drive commands are trajectories(.traj files) located in deploy/choreo. We schedule these by passing in the name of the trajectory as a string.

AG6GR commented on src/main/java/frc/robot/systems/ClimberFSMSystem.java:
```diff
@@ -161,120 +170,78 @@ public void update(TeleopInput input) {
-
-			case EXTENDED:
+			case IDLE:
+				// copies MANUAL's state transitions
```
If that's the case, do you still need an IDLE state?

Adbritto replied:
We need IDLE since that is out default state and MANUAL can be activated on button hold

AG6GR commented on src/main/java/frc/robot/systems/LEDFSMSystem.java:
```diff
@@ -0,0 +1,285 @@
+	 * Handle behavior in REEF_ALIGNED.
+	 */
+	private void handleReefAlignedState() {
+		led.setReefAlignColor();
```
Do we really need separate setXYZColor functions? They're only getting called from the state handlers right?

Adbritto replied:
Not really, will update to set color directly in the fsm

Adbritto committed:
resolve pr comments for mech

cleanup unused code
led directly sets colors instead of needing BlinkinLED Class
rename drive controller button getters
finalize all ports and ids
add code to download driver gui from bot

beranki committed:
remove tag sorting on java side

beranki committed:
Merge branch 'dev/week-7/sw-drive/cv-local' of https://github.com/Tino-FRC-2473/FRC2025 into dev/week-7/sw-drive/cv-local

beranki committed:
adding timer to outtake coral cmd

Adbritto committed:
remove redundant timer reset

Adbritto committed:
update vendordeps

pragisharma committed:
finalize ds layout

pragisharma committed:
resolve pr comment

queerboi commented on src/main/java/frc/robot/systems/ElevatorFSMSystem.java:
```diff
@@ -210,6 +212,17 @@ public void updateLogging() {
+	 * Is elevator at L4 boolean accessor.
+	 * @return whether or not elevator is at L4.
+	 */
+	public boolean isElevatorAtL4() {
```
Can we rework this so that it returns if the elevator is **too high**, not just at the very top at L4? this way we can damp speed even at L3 or something.

pragisharma committed:
tuning and testing

pragisharma committed:
Merge branch 'main' into dev/week-7/sw-drive/cv-local

pragisharma committed:
fix javadoc
